### PR TITLE
fix two problems will let fuzzing quits eariler

### PR DESCRIPTION
**Pull request by lum1n0us**

### CODE REVIEW

**Comment by lum1n0us** [comment]
depends on #3464

File: core/iwasm/interpreter/wasm_loader.c [hunk 1]
```diff
@@ -5280,18 +5281,25 @@ calculate_global_data_offset(WASMModule *module)
 #if WASM_ENABLE_FAST_JIT != 0
         import_global->data_offset = data_offset;
 #endif
-        data_offset += wasm_value_type_size(import_global->type.val_type);
+        int32 type_size = wasm_value_type_size(import_global->type.val_type);
```
[the remark this answers is not in the document]
Comment (lum1n0us) [reply core/iwasm/interpreter/wasm_loader.c:5284]: the failure happens during `wasm_value_type_size()`. So we can't check the type after L5284

[the remark this answers is not in the document]
Comment (lum1n0us) [reply core/iwasm/interpreter/wasm_loader.c:5284]: Do u mean to check `declare_type` after L3055?

[the remark this answers is not in the document]
Comment (wenyongh) [reply core/iwasm/interpreter/wasm_loader.c:5284]: Yes, had better check declare it after L3055.

[the remark this answers is not in the document]
Comment (lum1n0us) [reply core/iwasm/interpreter/wasm_loader.c:5284]: Just realized [`static int is_value_type()`](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_loader.c#L325C1-L344C2) might match most of my requirement(e.g. `is_value_type() || is_packed_type()` ). But it is lack of `VALUE_TYPE_ANY` and `VALUE_TYPE_VOID`. Should I add them into `is_value_type()` or just create a new function to validate, like `validate_global_value_type()`

[the remark this answers is not in the document]
Comment (wenyongh) [reply core/iwasm/interpreter/wasm_loader.c:5284]: Yes, use `is_value_type()` should be good, like what `resolve_value_type` does when GC is enabled.
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_loader.c#L1464-L1468

resolve_value_type is called when resolving the global type, table elem type, function param/result types when GC is enabled.

 And don't need to add `|| is_packed_type()`, the packed type i8/i16 is introduced by GC proposal and is used only in struct field type and array element type, the global type should not be i8 or i16.

File: core/config.h [hunk 2]
```diff
@@ -663,4 +663,17 @@
 #define WASM_MEM_ALLOC_WITH_USAGE 0
 #endif
 
+#ifndef WASM_ENABLE_FUZZ_TEST
+#define WASM_ENABLE_FUZZ_TEST 0
+#endif
+
+#ifndef WASM_MEM_ALLOC_MAX_SIZE
+#ifdef WASM_ENABLE_FUZZ_TEST
```
Comment (wenyongh) [inline core/config.h:671]: Should be `#if WASM_ENABLE_FUZZ_TEST == 0`

File: core/iwasm/interpreter/wasm_loader.c [hunk 3]
```diff
@@ -3052,7 +3053,10 @@ load_global_import(const uint8 **p_buf, const uint8 *buf_end,
 
 #if WASM_ENABLE_GC == 0
     CHECK_BUF(p, p_end, 2);
+    /* global type */
     declare_type = read_uint8(p);
+    if (!is_value_type(declare_type))
+        return false;
```
Comment (wenyongh) [inline core/iwasm/interpreter/wasm_loader.c:3059]: Had better `set_error_buf(error_buf, error_buf_size, "type mismatch");` before return false

File: core/iwasm/interpreter/wasm_loader.c [hunk 4]
```diff
@@ -4034,7 +4038,10 @@ load_global_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
         for (i = 0; i < global_count; i++, global++) {
 #if WASM_ENABLE_GC == 0
             CHECK_BUF(p, p_end, 2);
+            /* global type */
             global->type.val_type = read_uint8(p);
+            if (!is_value_type(global->type.val_type))
+                return false;
```
Comment (wenyongh) [inline core/iwasm/interpreter/wasm_loader.c:4044]: same as above

**Review by wenyongh — APPROVED** [review]
LGTM

**Pull request merged**